src/averageAges.js

```diff
@@ -1,61 +1,72 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filterMenbyCent = people.filter(men =>
+    men.sex === 'm' && Math.ceil(men.died / 100) === century);
+  const resMen = century
+    ? filterMenbyCent
+    : people.filter(men => men.sex === 'm');
+
+  const averageMen = resMen.reduce((prev, person) =>
+    (prev + (person.died - person.born)), 0) / resMen.length;
+
+  // console.log('averageMen :', averageMen);
+
+  return averageMen;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average ave of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find the other who mention
- * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWombyChild = people.filter(women =>
```

Comment:
'filterWombyChild' - name for some function, change it (womenWithChild for example)